libs/content/workspace-ffi/src/lib.rs

```diff
@@ -152,7 +139,53 @@ impl<'window> WgpuWorkspace<'window> {
             }
         };
 
-        out
+        #[cfg(not(target_os = "android"))]
+        let copied_text = {
+            let ct = full_output.platform_output.copied_text;
+            if ct.is_empty() {
+                std::ptr::null_mut()
+            } else {
+                CString::new(ct).unwrap().into_raw()
+            }
+        };
+
+        #[cfg(target_os = "android")]
+        let copied_text = full_output.platform_output.copied_text;
+
+        #[cfg(not(target_os = "android"))]
+        let url_opened = {
+            let url = full_output.platform_output.open_url;
+            if let Some(url) = url {
+                CString::new(url.url).unwrap().into_raw()
+            } else {
+                std::ptr::null_mut()
+            }
+        };
+
+        #[cfg(target_os = "android")]
+        let url_opened = full_output
+            .platform_output
+            .open_url
+            .map(|url| url.url)
+            .unwrap_or_default();
+
+        let cursor = full_output.platform_output.cursor_icon.into();
+
+        let virtual_keyboard_shown = self.context.pop_virtual_keyboard_shown();
+        let virtual_keyboard_shown_set = virtual_keyboard_shown.is_some();
+        let virtual_keyboard_shown_val = virtual_keyboard_shown.unwrap_or_default();
+
+        // todo: export context_menu_pos via self.context.pop_context_menu_pos()
+
+        Output {
```

Comment:
assembling this at the end instead of mutating it along the way is a style change that helps me make sure I didn't forget to populate something